src/components/DataViewer/SampleDataViewerWithButton.tsx

```diff
@@ -0,0 +1,49 @@
+"use client";
+import React, { useState } from "react";
+import DataViewer from "@/components/DataViewer/DataViewer";
+// import PhoneIcon from "../Icons/PhoneIcon";
+import styled from "styled-components";
+
+interface ViewerProps {
+    data: { [key: string]: string | number | null };
+}
+
+const Header = styled.span`
+    display: flex;
+    align-items: center;
+`;
+
+const Title = styled.span`
+    margin-left: 0.5rem;
+`;
+
+const SampleDataViewerWithButton: React.FC<ViewerProps> = ({ data }) => {
+    const [viewerIsOpen, setViewerIsOpen] = useState(false);
+
+    const openModal: () => void = () => {
+        setViewerIsOpen(true);
+    };
+
+    const closeModal: () => void = () => {
+        setViewerIsOpen(false);
+    };
+
+    return (
+        <>
+            <button onClick={openModal}>Open</button>
```

Comment:
might be worth using `Button` (the styled one) from the shared components folder

Reply:
The current button takes a href which is not what we want as we need to use the onClick. I think we could change it once theres a better button implemented in a separate ticket.